core/src/test/scala/unit/kafka/admin/TopicCommandWithAdminClientTest.scala

```diff
@@ -223,8 +224,20 @@ class TopicCommandWithAdminClientTest extends KafkaServerTestHarness with Loggin
     createAndWaitTopic(createOpts)
 
     // try to re-create the topic
-    intercept[IllegalArgumentException] {
+    intercept[TopicExistsException] {
+      topicService.createTopic(createOpts)
+    }
+  }
+
+  @Test
+  def testCreateWhenAlreadyExistsWithIfNotExists(): Unit = {
+    val createOpts = new TopicCommandOptions(Array("--topic", testTopicName, "--if-not-exists"))
+    createAndWaitTopic(createOpts)
+
+    try {
       topicService.createTopic(createOpts)
+    } catch {
```

Comment:
Hmm.  I don't see why these catch blocks are needed.  All we're doing here is just failing when we get an exception.  But an exception propagating to the top level of the test function fails the test anyway.  What am I missing?

Reply:
I think the distinction is between whether you want the test to error or fail. Here I wanted the tests to fail with a contextual error by calling `fail()`. Also makes the test read more explicitly.. i.e it says we only treat the specific caught exception as the expected failure by calling `fail()`, while any other exception will just error out the test..

Reply:
I guess personally I find it confusing since we're catching a particular type of exception specifically, but then just doing what we would have done anyway if we hadn't caught it (failing).  Also the failure message "Alter topic should not throw exception" implies that it is catching all exceptions, but actually only catching some that the code could throw.  So doubly confusing.